### ISSUE TITLE
Possible fix for SC2015

### ISSUE THREAD

**Issue by avidseeker** [koalaman/shellcheck#3291]
SC 2015 says:

> It's common to use `A && B` to run `B` when `A` is true, and `A || C` to run `C` when `A` is false.
> 
> However, combining them into `A && B || C` is not the same as `if A then B else C`.
> 
> In this case, if `A` is true but `B` is false, `C` will run.

But this could be fixed with either

```sh
A && { B; return; } || C
```

or

```sh
A && { B; exit; } || C
```

Ofc these aren't equivalent to using the proper if else statement, but shellcheck shouldn't complain when this workaround is used.

It could also be fixed using a subshell, in which case it'd seem to be equivalent ot if-else:

```sh
A && ( B; exit; ) || C
```
